File: yh_connect/src/yh_connect_sub.cpp
#include "ros/ros.h"
#include "std_msgs/Float32.h"

void msgCallback(const std_msgs::Float32::ConstPtr& msg){
    ROS_INFO("%f", msg->data);
}

int main(int argc, char** argv){
    
    ros::init(argc, argv, "yh_connect_sub");
    ros::NodeHandle nh;

    // 서브스크라이버 선언
    ros::Subscriber sub = nh.subscribe<std_msgs::Float32>("yh_connect_float", 10, msgCallback);

    ros::spin();

    return 0;
}

File: yh_connect/scripts/yh_connect_sub.py
#!/usr/bin/python3
#-*- coding:utf-8 -*-

import rospy
from std_msgs.msg import Float32

class ConnectSub:
    """
    ConnectPub Class
    @param 없음
    """
    def __init__(self):
        self.sub = rospy.Subscriber("yh_connect_float", Float32, self.msgCallback)
    
    def msgCallback(self, msg):
        rospy.loginfo(msg.data)

if __name__ =="__main__":
    rospy.init_node("yh_connect_sub") # 노드 이름 초기화
    connect_sub = ConnectSub()
    rospy.spin()
